src/libraries/System.Private.CoreLib/src/System/Numerics/Quaternion.cs

```diff
@@ -188,31 +188,45 @@ public static Quaternion Identity
         /// <remarks>The <see cref="Quaternion.op_Multiply" /> method defines the operation of the multiplication operator for <see cref="Quaternion" /> objects.</remarks>
         public static Quaternion operator *(Quaternion value1, Quaternion value2)
         {
-            Quaternion ans;
-
-            float q1x = value1.X;
-            float q1y = value1.Y;
-            float q1z = value1.Z;
-            float q1w = value1.W;
-
-            float q2x = value2.X;
-            float q2y = value2.Y;
-            float q2z = value2.Z;
-            float q2w = value2.W;
-
-            // cross(av, bv)
-            float cx = q1y * q2z - q1z * q2y;
-            float cy = q1z * q2x - q1x * q2z;
-            float cz = q1x * q2y - q1y * q2x;
-
-            float dot = q1x * q2x + q1y * q2y + q1z * q2z;
-
-            ans.X = q1x * q2w + q2x * q1w + cx;
-            ans.Y = q1y * q2w + q2y * q1w + cy;
-            ans.Z = q1z * q2w + q2z * q1w + cz;
-            ans.W = q1w * q2w - dot;
-
-            return ans;
+            if (Vector128.IsHardwareAccelerated)
+            {
+                var left = value1.AsVector128();
+                var right = value2.AsVector128();
+
+                var result = right * left.GetElementUnsafe(3);
+                result += (Vector128.Shuffle(right, Vector128.Create(3, 2, 1, 0)) * left.GetElementUnsafe(0)) * Vector128.Create(+1.0f, -1.0f, +1.0f, -1.0f);
+                result += (Vector128.Shuffle(right, Vector128.Create(2, 3, 0, 1)) * left.GetElementUnsafe(1)) * Vector128.Create(+1.0f, +1.0f, -1.0f, -1.0f);
+                result += (Vector128.Shuffle(right, Vector128.Create(1, 0, 3, 2)) * left.GetElementUnsafe(2)) * Vector128.Create(-1.0f, +1.0f, +1.0f, -1.0f);
+                return result.AsQuaternion();
```

Comment:
Looks like this will need to be
```suggestion
                return Unsafe.BitCast<Vector128<float>, Quaternion>(result);
```

I had *thought* I added `AsVector128` and `AsQuaternion` at the same time, but apparently I only added one. I'm going to open an API proposal to get them public and to cover the missing inverse conversions at the same time.

Reply:
My apologies, i just wanted to do a quick improvement and dont have a great development environment to catch this. I also indeed saw that the `AsQuaternion` existed as internal and assumed it was good. 

Should this PR use the bitcast, or would it be better to wait for the API proposal?

Reply:
I'd just use BitCast for now to unblock the PR and then we can go and fix it up later after the API becomes usable.